### DISCUSSION TITLE
How to check nccl is used

### DISCUSSION THREAD

**Discussion by wukong1992** [horovod/horovod#2771] [Q&A]
**Environment:**
I use horovod image:0.18.2-tf2.0.0-torch1.3.0-mxnet1.5.0-xm1.0.0-py3 , try to run, but there is not improvement, so, i suspect some thing is wrong ,and i find out there is no nccl log when task start. but when i check the hvd.nccl_buildt， it return true, so i want to know how to check i really use the nccl to  accelerate training

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Please ask your question here.

**Comment by tgaddair** [comment]
Hey @wukong1992, try generating a [timeline](https://github.com/horovod/horovod/blob/master/docs/timeline.rst).  If you see NCCL_Allreduce events, then you know NCCL is being used.  This will also help to understand where the bottleneck is coming from.

**Comment by wukong1992** [comment]
@tgaddair thank for your reply, I checked the timeline, only negotiate allreduce ,no nccl allreduce, does this mean nccl not used. command: mpirun -x HOROVOD_TIMELINE -allow-run-as-root -np 2 -bind-to none -map-by slot -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH  -x NCCL_SOCKET_IFNAME=eth0 -mca pml ob1 -mca btl ^openib -mca btl_tcp_if_include eth0 python ...
. timeline file:
[timeline.txt](https://github.com/horovod/horovod/files/5884365/timeline.txt)

**Comment by tgaddair** [comment]
Hey @wukong1992, I do see `NCCL_ALLREDUCE` calls in your timeline, so NCCL is being used.  However, if you're seeing long negotiation periods, it could mean that network or workload imbalance is an issue.
